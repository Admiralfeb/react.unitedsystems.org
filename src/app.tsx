import { ThemeProvider, CssBaseline } from '@material-ui/core';
import { darkTheme } from './theme';
import { createBrowserHistory } from 'history';

import { Router } from 'react-router-dom';
import { NavbarComponent } from 'components/navbar';
import { AppRoutes } from 'routes/app.routes';
import './app.css';
import { RealmAppProvider, RealmApolloProvider } from 'providers';
import { SnackbarProvider } from 'notistack';

const realmID: string = 'usc-tbmbi';

function App() {
  const history = createBrowserHistory({ basename: process.env.PUBLIC_URL });

  return (
    <RealmAppProvider appId={realmID}>
      <RealmApolloProvider>
        <SnackbarProvider maxSnack={3}>
          <ThemeProvider theme={darkTheme}>
            <CssBaseline />
            <div className="content">
              <Router history={history}>
                <NavbarComponent />
                <AppRoutes />
              </Router>
            </div>
          </ThemeProvider>
        </SnackbarProvider>
      </RealmApolloProvider>
    </RealmAppProvider>
  );
}

export default App;
