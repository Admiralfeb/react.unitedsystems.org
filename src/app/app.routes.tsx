import { Switch, Route, Redirect } from 'react-router-dom';
import { Home, Join, NotFound } from './components';
import { Information } from './information/information.routes';

export const AppRoutes = () => {
  return (
    <Switch>
      <Redirect from='/' to='/home' exact />
      <Route path='/home' exact>
        <Home />
      </Route>
      <Route path='/join'>
        <Join />
      </Route>
      <Route path='/information'>
        <Information />
      </Route>
      <Route path='*'>
        <NotFound />
      </Route>
    </Switch>
  );
};
