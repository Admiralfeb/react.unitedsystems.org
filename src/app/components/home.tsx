import { makeStyles, Paper, Typography } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  root: {},
  header: {
    textAlign: 'center',
  },
  paper: {
    textAlign: 'center',
    width: '80%',
    margin: 'auto',
    padding: 5,
    paddingBottom: 10,
    marginBottom: 5,
  },
  latin: {
    display: 'flex',
    flexDirection: 'column',
    textAlign: 'center',
    border: '1px solid white',
    borderRadius: 10,
    width: 300,
    margin: 'auto',
    marginBottom: 10,
  },
  cursive: {
    fontFamily: 'Brush Script MT, Brush Script Std, cursive',
  },
  subtitle: {
    flex: '0 0',
  },
  video: {
    width: '100%',
    padding: 0,
  },
});

export const Home = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Typography variant='h3' className={classes.header}>
        United Systems Cooperative
      </Typography>
      <Paper className={classes.paper}>
        <div className={classes.latin}>
          <Typography className={classes.cursive} variant='h4'>
            Ad Astra Per Aspera
          </Typography>
          <Typography variant='subtitle2' className={classes.subtitle}>
            Through Hardships to the Stars
          </Typography>
        </div>
        <Typography>
          Established in 3306, the United Systems Cooperative is a tight-knit
          organization of independent pilots operating across the galaxy, both
          in human space, and the uncharted sectors beyond. Little is known
          about this groups motives and plans, though many conflicting reports
          are offered by systems across inhabited space. The one thing the
          reports agree on is that members of this group are fiercely loyal to
          one another, and to the organization as a whole. The United Systems
          Cooperative seeks to create a family where everyone is welcome, no
          matter their background, their allegiances, etc. To create a family
          where everyone can get the help they need, make lasting friendships,
          and have new brothers and sisters to fly with them through the black!
        </Typography>
        <video
          controls
          className={classes.video}
          poster='https://dl.dropbox.com/s/nws07775923ta6z/vidImg.png?dl=0'>
          <source
            src='https://dl.dropbox.com/s/h0n88hy5pgtf06n/about_us.mp4?dl=0'
            type='video/mp4'
          />
          "Your browser does not support this video"
        </video>
      </Paper>
    </div>
  );
};
