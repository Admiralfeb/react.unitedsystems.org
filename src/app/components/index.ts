export * from '../information/components/about/about';
export * from './home';
export * from './join';
export * from './loading';
export * from './notFound';
export * from './underConstruction';
