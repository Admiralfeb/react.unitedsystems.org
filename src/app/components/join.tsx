import {
  Link,
  makeStyles,
  Paper,
  Typography,
  useMediaQuery,
} from '@material-ui/core';
import { NavLink } from 'react-router-dom';

const useStyles = makeStyles({
  root: {
    height: '100%',
  },
  header: {
    textAlign: 'center',
  },
  paper: {
    textAlign: 'center',
    width: '80%',
    margin: 'auto',
    padding: 5,
    paddingBottom: 10,
    marginBottom: 5,
  },
  iframe: {
    width: '100%',
    height: '59vw',
  },
  step: {
    display: 'grid',
    gridTemplate: '1fr / 1fr 1fr',
  },
  right: {
    textAlign: 'right',
  },
  left: {
    textAlign: 'left',
  },
});

export const Join = () => {
  const classes = useStyles();
  const mediaQuery = useMediaQuery('(min-width:950px)');
  return (
    <div className={classes.root}>
      <Typography variant='h2' className={classes.header}>
        Join Us!
      </Typography>
      <Paper className={classes.paper}>
        <Typography>
          Step 1: Read the rules on the{' '}
          <Link to='/about' target='_blank' component={NavLink}>
            About Page
          </Link>
          .
        </Typography>
        {mediaQuery ? (
          <Typography>Step 2: Fill out the form below</Typography>
        ) : (
          <Typography>
            Step 2: Fill out the{' '}
            <Link
              href='https://docs.google.com/forms/d/e/1FAIpQLSfw7vIL8FY8ZGTF8YFjn3x-zGdhP5k9J6JALXAjXA7_Mm3-Rw/viewform?usp=sf_link'
              target='_blank'>
              USC Join Form
            </Link>
          </Typography>
        )}

        <Typography>
          Step 3: Join our{' '}
          <Link href='https://discord.gg/br8Fy2M6cv' target='_blank'>
            Discord
          </Link>
        </Typography>

        <Typography>
          Recommended: Find us in-game! (UCPC for PC | UCXB for Xbox | UCPS for
          PlayStation)
        </Typography>
        <Typography>
          Recommended: Join our{' '}
          <Link href='https://inara.cz/squadron/7028/' target='_blank'>
            Inara Squadron
          </Link>
        </Typography>
        {mediaQuery && (
          <iframe
            className={classes.iframe}
            title='USC Application Form'
            id='joinFormContainer'
            src='https://docs.google.com/forms/d/e/1FAIpQLSfw7vIL8FY8ZGTF8YFjn3x-zGdhP5k9J6JALXAjXA7_Mm3-Rw/viewform?embedded=true'>
            Loadingâ€¦
          </iframe>
        )}
        {/* <iframe
          className={classes.iframe}
          title='Discord Widget'
          id='embeddedDiscord'
          src='https://discord.com/widget?id=662439414152167434&amp;theme=dark'
          allowTransparency
          frameBorder='0'
          sandbox='allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts'></iframe> */}
      </Paper>
    </div>
  );
};
