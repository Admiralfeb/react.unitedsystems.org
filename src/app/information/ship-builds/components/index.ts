export * from './buildItem';
export * from './builds';
export * from './query';
export * from './selection';
