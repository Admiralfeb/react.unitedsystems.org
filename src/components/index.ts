export * from './home';
export * from '../routes/information.routes';
export * from './join';
export * from './joinForm';
export * from './notFound';
