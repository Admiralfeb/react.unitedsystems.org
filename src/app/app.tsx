import React from 'react';

import { ThemeProvider, CssBaseline } from '@material-ui/core';
import { darkTheme } from './theme';
import { createBrowserHistory } from 'history';

import { Router } from 'react-router-dom';
import { NavbarComponent } from './navbar';
import { AppRoutes } from './app.routes';
import './app.css';
import { RealmAppProvider } from './providers/realmAppProvider';
import { RealmApolloProvider } from './providers/realmApolloProvider';

const realmID: string = 'usc-tbmbi';

function App() {
  const history = createBrowserHistory({ basename: process.env.PUBLIC_URL });

  return (
    <RealmAppProvider appId={realmID}>
      <RealmApolloProvider>
        <ThemeProvider theme={darkTheme}>
          <CssBaseline />
          <div className='content'>
            <Router history={history}>
              <NavbarComponent />
              <AppRoutes />
            </Router>
          </div>
        </ThemeProvider>
      </RealmApolloProvider>
    </RealmAppProvider>
  );
}

export default App;
