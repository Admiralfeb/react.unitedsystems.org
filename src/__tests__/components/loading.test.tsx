import { cleanup, render, RenderResult } from '@testing-library/react';
import { Loading } from '../../app/components';

let componentBody: RenderResult;

describe('Join', () => {
  beforeEach(() => {
    componentBody = render(<Loading />);
  });

  afterEach(() => {
    cleanup();
  });

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
