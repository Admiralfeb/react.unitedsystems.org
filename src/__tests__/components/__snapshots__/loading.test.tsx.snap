// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-1"
    >
      <h3>
        Loading
      </h3>
      <img
        alt="Loading"
        id="imgLoading"
        src="EDLoader1.svg"
        width="300"
      />
    </div>
  </div>
</body>
`;
